packages/versioning/src/versioning.ts

```diff
@@ -534,15 +549,102 @@ export function getVersions(p: Program, t: Type): [Namespace, VersionMap] | [] {
 // these decorators take a `versionSource` parameter because not all types can walk up to
 // the containing namespace. Model properties, for example.
 export function addedAfter(p: Program, type: Type, version: ObjectType) {
+  reportDeprecated(p, "Deprecated: addedAfter is deprecated. Use existsOnVersion instead.", type);
   const appliesAt = appliesAtVersion(getAddedOn, p, type, version);
   return appliesAt === null ? false : !appliesAt;
 }
 
 export function removedOnOrBefore(p: Program, type: Type, version: ObjectType) {
+  reportDeprecated(
+    p,
+    "Deprecated: removedOnOrBefore is deprecated. Use existsOnVersion instead.",
+    type
+  );
   const appliesAt = appliesAtVersion(getRemovedOn, p, type, version);
   return appliesAt === null ? false : appliesAt;
 }
 
+function getSortedVersions(p: Program, t: Type): Version[] | undefined {
+  const [namespace, _] = getVersions(p, t);
+  if (namespace === undefined) return undefined;
+
+  const allVersions = getVersion(p, namespace)?.getVersions();
+  return allVersions?.sort((a, b) => a.index - b.index);
+}
+
+function getFirst(arr: Version[]): Version | undefined {
+  if (!arr.length) return undefined;
+  if (arr.length > 1) {
+    // TODO: Error?
+  }
+  return arr[0];
+}
+
+export enum Availability {
+  Unavailable = "Unavailable",
+  Added = "Added",
+  Available = "Available",
+  Removed = "Removed",
+}
+
+/**
+ * Returns a map of version names and whether a given type is available in that version
+ * @param program Cadl program
+ * @param type Type to get the availability map for
+ * @returns the availability map for the type, if applicable
+ */
+export function getAvailabilityMap(
+  program: Program,
+  type: Type
+): Map<string, Availability> | undefined {
+  const avail = new Map<string, Availability>();
+
+  const allVersions = getSortedVersions(program, type);
+  // if unversioned then everything exists
+  if (allVersions === undefined) return undefined;
+
+  let added = program.stateMap(addedOnKey).get(type) as Version[];
```

Comment:
Can maybe make helper functions(not exported) to access the stateMap of added and removed

Reply:
Or should they be exported?  That's why I didn't make helper functions, but right now, `getAddedOn` just says it's deprecated. It doesn't actually give an alternative. Should I expose the analogous correct helper?

Reply:
its mostly so the code is cleaner and you don't have to do the `as ...` type casting inside a function logic.

Reply:
I guess if folks want an updated helper we can add them in the future as a non-breaking change.

Reply:
Added these helpers but didn't export them.